packages/core/src/blocks/defaultBlockTypeGuards.ts

```diff
@@ -0,0 +1,156 @@
+import {
```

Comment:
Yeah I guess that's fair, I was thinking if someone wanted to e.g. replace the heading block with their own implementation you would need to do a deep equals, but since the props have to be the same that's not really a realistic use case. So reference check should be fine